models/demos/wormhole/llama31_8b/tt/model_config.py

```diff
@@ -242,18 +268,30 @@ def __init__(self, device, instruct=False):
                 fuse_batch=seq_len <= 2048,
             )
 
-            # FIXME: This configuration avoids di/dt non-deterministic hangs. Issue #11354
-            self.model_config["OUTPUT_MM_PROGCFG"] = ttnn.MatmulMultiCoreReuseMultiCast1DProgramConfig(
-                compute_with_storage_grid_size=(7, 8),
-                in0_block_w=1,
-                per_core_M=1,
-                per_core_N=72,  # vocab size = 128k = 4008 tiles. 4008/56cores = 72
-                out_subblock_h=1,
-                out_subblock_w=1,
-                fuse_batch=True,
-                fused_activation=None,
-                mcast_in0=True,
-            )
+            if os.getenv("DI_DT_WORKAROUND") == "1":
+                self.model_config["OUTPUT_MM_PROGCFG"] = ttnn.MatmulMultiCoreReuseMultiCast1DProgramConfig(
+                    compute_with_storage_grid_size=(7, 8),
+                    in0_block_w=1,
+                    per_core_M=1,
+                    per_core_N=72,  # vocab size = 128k = 4008 tiles. 4008/56cores = 72
+                    out_subblock_h=1,
+                    out_subblock_w=1,
+                    fuse_batch=True,
+                    fused_activation=None,
+                    mcast_in0=True,
+                )
+            else:
```

Comment:
Just out of curiosity, did you test the perf with this version vs. didt workaround? Both versions should be using 56 cores, right?

Reply:
Yes I did, substantial improvement with `in0_block_w=2` vs 1 and `out_subblock_w=4` vs 1. The non-di/dt version uses 64 cores because why not, but that's not doing the heavy lifting perf-wise here.